set-up-location-driven-geotriggers/src/main/java/com/esri/arcgismaps/sample/setuplocationdrivengeotriggers/MainActivity.kt

```diff
@@ -0,0 +1,374 @@
+            sectionGeotrigger -> {
+                // update the section button's onClickListener
+                // to show a new FeatureViewFragment
+                sectionButton.setOnClickListener { showFeatureViewFragment(fenceFeature) }
+                // update the section button text to the feature name
+                sectionButton.text = fenceFeatureName
+                // enable the button
+                sectionButton.isEnabled = true
+            }
+            // or a point of interest geo trigger
+            poiGeotrigger -> {
+                // add it to the stored list
+                poiList.add(fenceFeature)
+                // notify the list adapter to refresh its recycler views
+                poiListAdapter.notifyItemInserted(poiList.lastIndex)
+            }
+        }
+    }
+
+    /**
+     * Removes the ArcGISFeature with the given [fenceFeatureName] and corresponding
+     * [geotriggerType] from the current UI state and refreshes the UI.
+     */
+    private fun removeFeatureInformation(fenceFeatureName: String, geotriggerType: String) {
+        // check the type of geotrigger
+        when (geotriggerType) {
+            sectionGeotrigger -> {
+                // if it's a section geo trigger,
+                // remove the section information and disable the button
+                sectionButton.text = "N/A"
+                sectionButton.isEnabled = false
+            }
+            poiGeotrigger -> {
+                // if it's a point of interest geotrigger
+                // find its index from the stored list
+                val index = poiList.indexOfFirst { feature ->
+                    feature.attributes["name"] == fenceFeatureName
+                }
+                if (index >= 0) {
+                    // if the feature exists remove it
+                    poiList.removeAt(index)
+                    // notify the list adapter to refresh its recycler views
+                    poiListAdapter.notifyItemRemoved(index)
+                }
+            }
+        }
+    }
+
+    /**
+     * Creates and shows a new FeatureViewFragment using the given [feature]
+     */
+    private fun showFeatureViewFragment(feature: ArcGISFeature) {
+        // stop the simulated data source
+        stopSimulatedDataSource(false)
+        // create a new FeatureViewFragment
+        val featureViewFragment = FeatureViewFragment(feature) {
+            // set it's onDismissedListener to
+            // resume the simulated data source
+            startSimulatedDataSource(false)
+        }
+        // show the fragment
+        featureViewFragment.show(supportFragmentManager, "FeatureViewFragment")
+    }
+
+    /**
+     * Starts the simulated data source and shows a status toast if [showAlert] is true.
+     * The data source is resumed from its previous location if stopped before.
+     */
+    private fun startSimulatedDataSource(showAlert: Boolean = true) = lifecycleScope.launch {
```

Comment:
I'm not such a fan of these default arguments (for this method and the one below) because the sample passes true and false equally often so I don't think they are really default values, and it would make more sense to specify explicitly for each call.